web-services/map-reduce-embedded/src/main/java/datawave/security/system/EmbeddedCallerPrincipalProducer.java

```diff
@@ -0,0 +1,38 @@
+package datawave.security.system;
+
+import datawave.security.authorization.DatawavePrincipal;
+import org.infinispan.commons.util.Base64;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import javax.enterprise.inject.Produces;
+import javax.interceptor.Interceptor;
+
+/**
+ * Caller principal producer supplied just for Embedded mode (e.g., inside of MapReduce jars). This archive should not be included for normal web applications.
+ */
+@Alternative
+@Priority(Interceptor.Priority.APPLICATION)
+@ApplicationScoped
+public class EmbeddedCallerPrincipalProducer {
+    private boolean initialized = false;
+    private DatawavePrincipal callerPrincipal;
+    
+    @Produces
+    @ServerPrincipal
+    public DatawavePrincipal produceCallerPrincipal() {
+        if (!initialized) {
+            initializeCallerPrincipal();
+            initialized = true;
+        }
+        return callerPrincipal;
+    }
+    
+    private void initializeCallerPrincipal() {
```

Comment:
Not a big deal, but since this is an ApplicationScoped bean, I believe you could just do this initialization in a constructor and then you wouldn't need the lazy init code in `produceCallerPrincipal()`.